flink-libraries/flink-table/src/main/scala/org/apache/flink/table/plan/nodes/datastream/DataStreamSort.scala

```diff
@@ -0,0 +1,169 @@
+      description + sort.getId())
+  }
+
+  override def toString: String = {
+    s"Sort($sort)" +
+      " on fields: (${sort.collation.getFieldCollations})"
+  }
+
+  override def explainTerms(pw: RelWriter): RelWriter = {
+    super.explainTerms(pw)
+      .item("aggregate", sort)
+      .item("sort fields",sort.collation.getFieldCollations)
+      .itemIf("offset", sort.offset, sort.offset!=null)
+      .itemIf("fetch", sort.fetch, sort.fetch!=null)
+      .item("input", inputNode)
+  }
+
+  override def translateToPlan(tableEnv: StreamTableEnvironment): DataStream[Row] = {
+
+    val inputDS = getInput.asInstanceOf[DataStreamRel].translateToPlan(tableEnv)
+    
+    //need to identify time between others order fields. Time needs to be first sort element
+    val timeType = SortUtil.getTimeType(sort,inputType)
+    
+    //time ordering needs to be ascending
+    if (SortUtil.getTimeDirection(sort) != Direction.ASCENDING) {
+      throw new TableException("SQL/Table supports only ascending time ordering")
+    }
+      
+     
+    val (offset,fetch) = (sort.offset,sort.fetch)
+    
+    //enable to extend for other types of aggregates that will not be implemented in a window
+    timeType match {
+        case _: ProcTimeType =>
+            (offset,fetch) match {
+              case (o:Any,f:Any)  => null             // offset and fetch needs retraction
+              case (_,f:Any) => null                  // offset needs retraction
+              case (o:Any,_) => null                  // fetch needs retraction
+              case _ => createSortProcTime(inputDS)   //sort can be done with/without retraction
+            }
+        case _: RowTimeType =>
+          throw new TableException("SQL/Table does not support sort on row time")
+        case _ =>
+          throw new TableException("SQL/Table needs to have sort on time as first sort element")
+    }
+    
+  }
+
+  /**
+   * Create Sort logic based on processing time
+   */
+  def createSortProcTime(
+    inputDS: DataStream[Row]): DataStream[Row] = {
+
+          
+    // get the output types
+    //Sort does not do project.= Hence it will output also the ordering proctime field
+    //[TODO]Do we need to drop some of the ordering fields? (implement a projection logic?
+    val rowTypeInfo = FlinkTypeFactory.toInternalRowTypeInfo(getRowType).asInstanceOf[RowTypeInfo]
+    
+    //if the order has secondary sorting fields in addition to the proctime
+    if( SortUtil.getSortFieldIndexList(sort).size > 1) {
+    
+      val processFunction = SortUtil.createProcTimeSortFunction(sort,inputType)
+      
+      inputDS
+            .keyBy(new NullByteKeySelector[Row])
+            .process(processFunction).setParallelism(1).setMaxParallelism(1)
+            .returns(rowTypeInfo)
+            .asInstanceOf[DataStream[Row]]
+    } else {
+      //if the order is done only on proctime we only need to forward the elements
+        inputDS.keyBy(new NullByteKeySelector[Row])
```

Comment:
@fhueske I thought we need it more to merge it into a global stream. Basically without the keyby there is no equivalence i think between using an identity map or using the actual sort (on a sort collection). What do you think?